root/usr/lib64/ruby/vendor_ruby/discovery/screen/info.rb

```diff
@@ -10,9 +10,13 @@ def screen_info action_proc, message, error_message, success_screen, failure_scr
   f = Newt::Form.new
   f.add(t)
   f.draw
+
+  # facter sometimes leaves messages on stdout, refresh before/after
+  Newt::Screen.refresh
+  result = action_proc.call
   Newt::Screen.refresh
 
-  if action_proc.call
+  if result
     success_screen
   else
```

Comment:
There is no good way to silence facter so this at least hides the output when it ocassionally print a message.